flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutorMemoryConfiguration.java

```diff
@@ -0,0 +1,333 @@
+
+	@JsonProperty(FIELD_NAME_METASPACE_MAX)
+	private final Long jvmMetaspaceMax;
+
+	@JsonProperty(FIELD_NAME_OVERHEAD_MIN)
+	private final Long jvmOverheadMin;
+	@JsonProperty(FIELD_NAME_OVERHEAD_MAX)
+	private final Long jvmOverheadMax;
+	@JsonProperty(FIELD_NAME_OVERHEAD_FRACTION)
+	private final Float jvmOverheadFraction;
+
+	@JsonProperty(FIELD_NAME_MEMORY_TOTAL)
+	private final Long totalProcessMemory;
+
+	private static Long getConfigurationValue(Configuration config, ConfigOption<? extends MemorySize> option) {
+		MemorySize memorySize = config.get(option);
+		return memorySize != null ? memorySize.getBytes() : null;
+	}
+
+	/**
+	 * Factory method for initializing a TaskExecutorMemoryConfiguration based on the passed Configuration.
+	 * @param config The Configuration used for initializing the TaskExecutorMemoryConfiguration.
+	 * @return The newly instantiated TaskExecutorMemoryConfiguration.
+	 */
+	public static TaskExecutorMemoryConfiguration create(Configuration config) {
+		return new TaskExecutorMemoryConfiguration(
+				getConfigurationValue(config, FRAMEWORK_HEAP_MEMORY),
+				getConfigurationValue(config, TASK_HEAP_MEMORY),
+				getConfigurationValue(config, FRAMEWORK_OFF_HEAP_MEMORY),
+				getConfigurationValue(config, TASK_OFF_HEAP_MEMORY),
+				getConfigurationValue(config, NETWORK_MEMORY_MIN),
+				getConfigurationValue(config, NETWORK_MEMORY_MAX),
+				config.getOptional(NETWORK_MEMORY_FRACTION).orElse(NETWORK_MEMORY_FRACTION.defaultValue()),
+				getConfigurationValue(config, MANAGED_MEMORY_SIZE),
+				config.getOptional(MANAGED_MEMORY_FRACTION).orElse(MANAGED_MEMORY_FRACTION.defaultValue()),
+				getConfigurationValue(config, JVM_METASPACE),
+				getConfigurationValue(config, JVM_OVERHEAD_MIN),
+				getConfigurationValue(config, JVM_OVERHEAD_MAX),
+				config.getOptional(JVM_OVERHEAD_FRACTION).orElse(JVM_OVERHEAD_FRACTION.defaultValue()),
+				getConfigurationValue(config, TOTAL_PROCESS_MEMORY)
+		);
+	}
+
+	@JsonCreator
+	public TaskExecutorMemoryConfiguration(
+			@JsonProperty(FIELD_NAME_FRAMEWORK_HEAP) Long frameworkHeap,
+			@JsonProperty(FIELD_NAME_TASK_HEAP) Long taskHeap,
+			@JsonProperty(FIELD_NAME_FRAMEWORK_OFFHEAP) Long frameworkOffHeap,
+			@JsonProperty(FIELD_NAME_TASK_OFFHEAP) Long taskOffHeap,
+			@JsonProperty(FIELD_NAME_NETWORK_MIN) Long networkMemoryMin,
+			@JsonProperty(FIELD_NAME_NETWORK_MAX) Long networkMemoryMax,
+			@JsonProperty(FIELD_NAME_NETWORK_FRACTION) Float networkMemoryFraction,
+			@JsonProperty(FIELD_NAME_MANAGED_TOTAL) Long managedMemoryTotal,
+			@JsonProperty(FIELD_NAME_MANAGED_FRACTION) Float managedMemoryFraction,
+			@JsonProperty(FIELD_NAME_METASPACE_MAX) Long jvmMetaspaceMax,
+			@JsonProperty(FIELD_NAME_OVERHEAD_MIN) Long jvmOverheadMin,
+			@JsonProperty(FIELD_NAME_OVERHEAD_MAX) Long jvmOverheadMax,
+			@JsonProperty(FIELD_NAME_OVERHEAD_FRACTION) Float jvmOverheadFraction,
+			@JsonProperty(FIELD_NAME_MEMORY_TOTAL) Long totalProcessMemory) {
```

Comment:
I've missed that one. But I'd rather provide the Flink memory configuration value as well. This way, it would only be calculated in one place (the backend) instead of two which is less error-prone. It's worth it since there's not not much code that needs to be added to expose it in the REST API. Any objections against that?

Reply:
The min/max values are merged together since they have the same value anyway after initialization. In case of a missing value, the default max value will be returned.